File: Sources/WeChatExporter/Models/ContactItem.swift
import Foundation

enum ContactKind: String, CaseIterable {
    case friend = "好友"
    case group = "群聊"
    case official = "公众号"

    var icon: String {
        switch self {
        case .friend: return "person.circle.fill"
        case .group: return "person.3.fill"
        case .official: return "megaphone.fill"
        }
    }
}

struct ContactItem: Identifiable, Hashable {
    let id: String
    let displayName: String
    let nickName: String
    let remark: String
    let kind: ContactKind
    let lastTime: String
    let lastTimestamp: Int
    let summary: String

    var subtitle: String {
        if summary.isEmpty { return lastTime }
        return "\(lastTime) · \(summary)"
    }
}

struct ExportResult: Identifiable {
    let id = UUID()
    let contact: ContactItem
    let messageCount: Int
    let outputURL: URL
}

File: windows/WeChatExporter.Windows/Models/ContactItem.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace WeChatExporter.Models;

public enum ContactKind
{
    Friend,
    Group,
    Official
}

public sealed class ContactItem : INotifyPropertyChanged
{
    private bool _isFavorite;

    public required string Id { get; init; }
    public required string DisplayName { get; init; }
    public required string NickName { get; init; }
    public required string Remark { get; init; }
    public required ContactKind Kind { get; init; }
    public required string LastTime { get; init; }
    public required long LastTimestamp { get; init; }
    public required string Summary { get; init; }

    public bool IsFavorite
    {
        get => _isFavorite;
        set
        {
            if (_isFavorite == value) return;
            _isFavorite = value;
            OnPropertyChanged();
            OnPropertyChanged(nameof(FavoriteGlyph));
        }
    }

    public string FavoriteGlyph => IsFavorite ? "★" : "☆";

    public string KindLabel => Kind switch
    {
        ContactKind.Group => "群聊",
        ContactKind.Official => "公众号",
        _ => "好友"
    };

    public string Subtitle => string.IsNullOrWhiteSpace(Summary)
        ? LastTime
        : $"{LastTime} · {Summary}";

    public event PropertyChangedEventHandler? PropertyChanged;

    private void OnPropertyChanged([CallerMemberName] string? name = null)
        => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
}
